dataframe-arrow/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/ArrowWriterImpl.kt

```diff
@@ -249,7 +249,7 @@ internal class ArrowWriterImpl(
         strictType: Boolean,
         strictNullable: Boolean,
     ): FieldVector {
-        val containNulls = (column == null || column.hasNulls())
+        val containNulls = (column == null || column.values().any { it == null})
```

Comment:
Please revert this line as it's less efficient. The `hasNulls()`/`isMarkedNullable` is set upon column creation and since columns are immutable we can safely assume all values are not null if `hasNulls() == false`. Maybe we could only run this check if `strictNullable == true`, wdyt @koperagen?

Reply:
OK, I have removed that commit on rebasing to new master. But we still can specify nullable type for not-null values and vice versa (look first message in #428, that test is reproduceable just now).

> Maybe we could only run this check if `strictNullable == true`

Sounds good, I can make this in separate PR. Or we should fix #428 totally in any another way (such as keeping "nullable" for types and "hasNulls" for values independently). You were agree that `hasNulls()` should only return true if there are actually nulls in the column.

Reply:
We're indeed still working on figuring out a good way to fix https://github.com/Kotlin/dataframe/issues/428 so that the cases you gave are not possible anymore. That way `hasNulls() == isMarkedNullable` at all times. The difficult part is finding a good balance between performance and safety. We don't want to have to scan an entire list of data if a user can ensure it has or has no nulls...